### PR TITLE
Add setting in config to control truncating field values.

### PR DESCRIPTION
**Pull request by j-bennet**
## Description

I resisted adding a config setting for `max_field_value`, but it keeps coming up. See:

https://github.com/dbcli/cli_helpers/pull/48#issuecomment-906602112
https://github.com/dbcli/pgcli/issues/1250

It's quite rare, but the reasonable default of truncating long fields doesn't work for some people. This allows people to have full control over it.

## Checklist

- [x] I've added this contribution to the `changelog.rst`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).
- [x] I installed pre-commit hooks (`pip install pre-commit && pre-commit install`), and ran `black` on my code.
- [x] Please squash merge this pull request (uncheck if you'd like us to merge as multiple commits)

### CODE REVIEW

File: tests/test_main.py [hunk 1]
```diff
@@ -128,11 +159,11 @@ def test_format_output_auto_expand():
     )
     table = [
         "Title",
-        "+---------+---------+",
-        "| head1   | head2   |",
-        "|---------+---------|",
-        "| abc     | def     |",
-        "+---------+---------+",
+        "+-------+-------+",
```
Comment (j-bennet) [inline tests/test_main.py:162]: Not sure why this changed - changes in tabulate? Seems unrelated to the new settings.

**Review by laixintao — APPROVED** [review]
LGTM
